### PR TITLE
[INLONG-3822][Sort] Add node relations definition to support transform

### LINKED ISSUE #3822: [Feature][Sort] Add LeftOuterJoinNodeRelationShip definition to support transform
### Description

Add LeftOuterJoinNodeRelationShip definition to support transform

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yunqingmoswu**
### Title Name: [INLONG-3822][Sort] Add node relations definition to support transform

Fixes #3822 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by healchow — COMMENTED** [review]
Please add some java docs, thanks.

**Review by healchow — APPROVED** [review]
LGTM

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/InnerJoinNodeRelationTest.java [hunk 1]
```diff
@@ -38,10 +38,15 @@
 import static org.junit.Assert.assertEquals;
 
 /**
- * Tests for{@link InnerJoinNodeRelationTest}
+ * Tests for {@link InnerJoinNodeRelationTest}
```
Comment (gong) [inline inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/InnerJoinNodeRelationTest.java:41]: `InnerJoinNodeRelationTest` should be InnerJoinNodeRelation

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/LeftOuterJoinNodeRelationTest.java [hunk 2]
```diff
@@ -38,10 +38,15 @@
 import static org.junit.Assert.assertEquals;
 
 /**
- * Tests for{@link LeftOuterJoinNodeRelationTest}
+ * Tests for {@link LeftOuterJoinNodeRelationTest}
```
Comment (gong) [inline inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/LeftOuterJoinNodeRelationTest.java:41]: `LeftOuterJoinNodeRelationTest` should be LeftOuterJoinNodeRelation

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/NodeRelationTest.java [hunk 3]
```diff
@@ -25,10 +25,15 @@
 import static org.junit.Assert.assertEquals;
 
 /**
- * Tests for{@link NodeRelationTest}
+ * Tests for {@link NodeRelationTest}
```
Comment (gong) [inline inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/NodeRelationTest.java:28]: `NodeRelationTest` should be NodeRelation

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/RightOuterJoinNodeRelationTest.java [hunk 4]
```diff
@@ -37,10 +37,15 @@
 import static org.junit.Assert.assertEquals;
 
 /**
- * Tests for{@link RightOuterJoinNodeRelationTest}
+ * Tests for {@link RightOuterJoinNodeRelationTest}
```
Comment (gong) [inline inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/RightOuterJoinNodeRelationTest.java:40]: `RightOuterJoinNodeRelationTest` should be RightOuterJoinNodeRelation

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/UnionNodeRelationTest.java [hunk 5]
```diff
@@ -26,10 +26,15 @@
 import static org.junit.Assert.assertEquals;
 
 /**
- * Tests for{@link UnionNodeRelationTest}
+ * Tests for {@link UnionNodeRelationTest}
```
Comment (gong) [inline inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/UnionNodeRelationTest.java:29]: `UnionNodeRelationTest` should be UnionNodeRelation

**Review by gong — APPROVED** [review]

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/UnionNodeRelationTest.java (hunk 5 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/UnionNodeRelationTest.java]: done

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/RightOuterJoinNodeRelationTest.java (hunk 4 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/RightOuterJoinNodeRelationTest.java]: done

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/NodeRelationTest.java (hunk 3 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/NodeRelationTest.java]: done

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/LeftOuterJoinNodeRelationTest.java (hunk 2 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/LeftOuterJoinNodeRelationTest.java]: done

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/InnerJoinNodeRelationTest.java (hunk 1 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/transformation/relation/InnerJoinNodeRelationTest.java]: done

**Review by EMsnap — APPROVED** [review]

**Pull request merged**